### ISSUE TITLE
Create TrainingPhase class and/or add "add_training_phase" method to ParametricDR

### ISSUE THREAD

**Issue by einbandi** [einbandi/paradime#1]
This class/method should allow users to set all the parameters required for a training phase. These are:

- number of epochs
- sampling method
- batch-wise relations (specified as a dict, so custom loss functions can acces them)
- loss function (which gets passed everything [model, global relations, batch-wise relations, batch] and accesses parts through keys)
